chart/templates/deployment.yaml

```diff
@@ -46,8 +46,32 @@ spec:
             - containerPort: 8080
             {{- end }}
           readinessProbe:
+            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
             exec:
-              command: ["ping", "-c", "3", "1.1.1.1"]
+              command:
+                - bash
+                - -c
+                - 'healthcheck.sh --probeURL={{ .Values.probeURL }} --proxyIP={{ .Values.listenOn }} --httpProxy={{- .Values.httpProxy.enabled }} --socksProxy={{- .Values.socksProxy.enabled }} --useHttps={{- .Values.useHttpsInProbeURL }}'
+          livenessProbe:
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            exec:
+              command:
+                - bash
+                - -c
+                - 'healthcheck.sh --probeURL={{ .Values.probeURL }} --proxyIP={{ .Values.listenOn }} --httpProxy={{- .Values.httpProxy.enabled }} --socksProxy={{- .Values.socksProxy.enabled }} --useHttps={{- .Values.useHttpsInProbeURL }}'
+          startupProbe:
+            failureThreshold: {{ .Values.startupProbe.failureThreshold }}
+            periodSeconds: {{ .Values.startupProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.startupProbe.timeoutSeconds }}
+            exec:
+              command:
+                - bash
+                - -c
+                - 'healthcheck.sh --probeURL={{ .Values.probeURL }} --proxyIP={{ .Values.listenOn }} --httpProxy={{- .Values.httpProxy.enabled }} --socksProxy={{- .Values.socksProxy.enabled }} --useHttps={{- .Values.useHttpsInProbeURL }}'
```

Comment:
Lot of potential for typos, probes are the same object, write a helper that generates the probe